gcovr/formats/cobertura/read.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding:utf-8 -*-
+
+#  ************************** Copyrights and license ***************************
+#
+# This file is part of gcovr 6.0+master, a parsing and reporting tool for gcov.
+# https://gcovr.com/en/stable
+#
+# _____________________________________________________________________________
+#
+# Copyright (c) 2013-2023 the gcovr authors
+# Copyright (c) 2013 Sandia Corporation.
+# Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+# the U.S. Government retains certain rights in this software.
+#
+# This software is distributed under the 3-clause BSD License.
+# For more information, see the README.rst file.
+#
+# ****************************************************************************
+
+import logging
+import os
+from glob import glob
+from lxml import etree
+
+from ...options import Options
+from ..gcov.read import apply_filter_include_exclude
+from ...coverage import (
+    BranchCoverage,
+    CovData,
+    FileCoverage,
+    LineCoverage,
+)
+from ...merging import (
+    get_merge_mode_from_options,
+    insert_branch_coverage,
+    insert_file_coverage,
+    insert_line_coverage,
+)
+
+LOGGER = logging.getLogger("gcovr")
+
+
+#
+#  Get coverage from already existing gcovr JSON files
+#
+def read_report(options: Options) -> CovData:
+    """merge a coverage from multiple reports in the format
+    compatible with Cobertura"""
+
+    if len(options.cobertura_add_tracefile) == 0:
+        return None
+
+    datafiles = set()
+
+    for trace_files_regex in options.cobertura_add_tracefile:
+        trace_files = glob(trace_files_regex, recursive=True)
+        if not trace_files:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n" "\tThe specified file does not exist."
+            )
+        else:
+            for trace_file in trace_files:
+                datafiles.add(os.path.normpath(trace_file))
+
+    covdata = dict()
+    for filename in datafiles:
+        LOGGER.debug(f"Processing XML file: {filename}")
+
+        try:
+            root = etree.parse(filename).getroot()
+        except Exception:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n"
+                f"\tThe file {filename} is not a valid XML file."
+            )
```

Comment:
Should the message of the original exception be added here? This is needed if there is a syntax error in the file.

Reply:
Yes, you're right. I just checked it for various types of errors (missing file/corrupt file) and the message from lxml contains the file name as well, so I can use that instead.